### PR TITLE
Remove unnecessary rules

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/project-system/issues/4247

This removes Xaml rule files for items where there is an `AvailableItemName` defined. Pay particular attention to https://github.com/dotnet/project-system/pull/5681/commits/f961ae45d3064907378f395e2dae6c28a0b3e9d7 which fixes a bug in the Fast Up to Date where it assumed that rule names match item types, but auto-generated rules in CPS do not follow that convention.

### CODE REVIEW

**Comment by davkean** [comment]
Does this remove the ability to have a "localized" sentenced cased name in the Build Actions for these generated ones?

**Comment by davidwengier** [comment]
Oh good point! Yes, there is no localization, the build actions are just shown in the dropdown. We'd have to invent something to extend `AvailableItemName` to allow it to supply a display name, and then support localization of same.

**Comment by davkean** [comment]
@davidwengier I think this would just bring us back to legacy compat which I'm okay with - though a followup to add English-based sentence casing would be good.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.State.cs [hunk 1]
```diff
@@ -314,8 +315,12 @@ private State()
 
                 bool itemsChanged = false;
 
-                foreach ((string itemType, IProjectChangeDescription projectChange) in sourceItemsUpdate.ProjectChanges)
+                foreach ((string schemaName, IProjectChangeDescription projectChange) in sourceItemsUpdate.ProjectChanges)
                 {
+                    // ProjectChanges is keyed by the rule name which is usually the same as the item type, but not always (eg, in auto-generated rules)
+                    string? itemType = projectCatalogSnapshot.NamedCatalogs[PropertyPageContexts.File].GetSchema(schemaName)?.DataSource.ItemType;
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.State.cs]: Is this indexer guaranteed to find its key?

Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.State.cs]: There is certainly lots of spots in code where it's assumed to exist, but an extra question mark here wouldn't hurt much. 👍

**Comment by davidwengier** [comment]
Any Microsofties who want to play around with this, the build is here: https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/VS/8cfd0934755bca0955616644ece196aaf317f0a7/74cd36dc-9833-f208-0707-184a070f7ef9%3Bbootstrappers/Enterprise/vs_enterprise.exe

**Comment by drewnoakes** [comment]
I was just reading the [CPS doc on extending XAML rules](https://github.com/microsoft/VSProjectSystem/blob/master/doc/extensibility/extending_rules.md) which explains that use of `OverrideMode="Replace"` _can only be performed on rules imported via MSBuild, and not by extension points._ Does this PR effect this?

**Comment by davidwengier** [comment]
This change would break extended or replaced rules, but I looked through GitHub and index and couldn't see any that are relevant. I think the risk is pretty low.

**Comment by davkean** [comment]
@davidwengier What's blocking this from going in?

**Comment by davidwengier** [comment]
The WPF team had some automation test failures from the build of this, late last year, but I haven't heard whether they'd investigated and were caused by this change or not.

I've been meaning to rebase it and get a new build for it just in case, so thanks for the reminder :)

**Comment by davidwengier** [comment]
New insertion is here: https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/227076

**Comment by davidwengier** [comment]
The automation tests all passed :tada:

**Pull request merged**